### PR TITLE
GPT4 integration

### PR DESCRIPTION
**Pull request by Ghastly2001**
i think it outlines a plan to evaluate the feasibility and cost-effectiveness of integrating the new GPT-4o model. The goal is to potentially reduce user costs by leveraging GPT-4o's purportedly lower pricing compared to GPT-4V.

### CODE REVIEW

File: skyvern/forge/sdk/api/llm/config_registry.py [hunk 1]
```diff
@@ -43,6 +46,17 @@ def get_config(cls, llm_key: str) -> LLMRouterConfig | LLMConfig:
...
+if SettingsManager.get_settings().ENABLE_GPT_4O:
+    LLMConfigRegistry.register_config(
+        "GPT_4O_MODEL_NAME",  # Replace with the actual model name
+        LLMConfig(
+            "gpt-4o-model-name",  # Replace with the actual model identifier
+            ["GPT_4O_API_KEY"],  # Use the API key defined in your settings
+            supports_vision=False,  # Set according to the model's capabilities
+            add_assistant_prefix=False,  # Adjust based on your naming conventions
+        ),
+    )
```
Comment (suchintan) [inline skyvern/forge/sdk/api/llm/config_registry.py:59]: I think the implementation is even simpler than this -- I think you can add a single line below the GPT-4V definition on line 77


    LLMConfigRegistry.register_config(
        "OPENAI_GPT4o", LLMConfig("gpt-4o", ["OPENAI_API_KEY"], supports_vision=True, add_assistant_prefix=False)
    )

File: skyvern/config.py [hunk 2]
```diff
@@ -62,7 +62,7 @@ class Settings(BaseSettings):
     # LLM Configuration #
     #####################
     # ACTIVE LLM PROVIDER
-    LLM_KEY: str = "OPENAI_GPT4V"
+    LLM_KEY: str = "DYNAMIC_LLM_SELECTION"
```
Comment (suchintan) [inline skyvern/config.py:65]:
```suggestion
    LLM_KEY: str = "OPENAI_GPT4o"
```

File: skyvern/config.py [hunk 3]
```diff
@@ -71,6 +71,7 @@ class Settings(BaseSettings):
     ENABLE_ANTHROPIC: bool = False
     ENABLE_AZURE: bool = False
     ENABLE_BEDROCK: bool = False
+    ENABLE_GPT_4O: bool = False
```
Comment (suchintan) [inline skyvern/config.py:74]: You don't need these changes! The change is easier than you think :)

**Comment by suchintan** [comment]
LFG!!

File: skyvern/config.py (hunk 3 above)
Comment (Ghastly2001) [reply skyvern/config.py:74]: Ahh,  i get it. Thanks for the feedback, tho. :)

Comment (suchintan) [reply skyvern/config.py:74]: @Ghastly2001 are you open to updating the PR? I'd love to get this merged:)

File: skyvern/forge/sdk/api/llm/config_registry.py [hunk 4]
```diff
@@ -43,6 +46,17 @@ def get_config(cls, llm_key: str) -> LLMRouterConfig | LLMConfig:
         return cls._configs[llm_key]
 
 
+# Check if GPT-4o is enabled and register it
+if SettingsManager.get_settings().ENABLE_GPT_4O:
```
Comment (suchintan) [inline skyvern/forge/sdk/api/llm/config_registry.py:50]: I think you can delete this entire block

File: skyvern/config.py [hunk 5]
```diff
@@ -81,6 +82,10 @@ class Settings(BaseSettings):
     AZURE_API_BASE: str | None = None
     AZURE_API_VERSION: str | None = None
 
+    # GPT-4o settings
+    GPT_4O_ENABLED: bool = False
```
Comment (suchintan) [inline skyvern/config.py:86]: I think you can delete these two settings

File: skyvern/config.py (hunk 3 above)
Comment (suchintan) [inline skyvern/config.py:74]: I think you can delete this setting

**Review by suchintan — COMMENTED** [review]
Almost there!! Looking really good

**Pull request merged**